thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/db/dao/AnomalyMergedResultDAO.java

```diff
@@ -26,29 +28,44 @@ public AnomalyMergedResultDAO() {
   @Transactional
   public List<AnomalyMergedResult> findByCollectionMetricDimensions(String collection,
       String metric, String dimensions) {
-    return super.findByParams(
-        ImmutableMap.of("collection", collection, "metric", metric, "dimensions", dimensions));
+    Map<String, Object> params = new HashMap<>();
+    params.put("collection", collection);
+    params.put("metric", metric);
+    params.put("dimensions", dimensions);
+    return super.findByParams(params);
   }
 
   @Transactional
   public AnomalyMergedResult findLatestByCollectionMetricDimensions(
       String collection, String metric, String dimensions) {
-    return getEntityManager()
-        .createQuery(FIND_BY_COLLECTION_METRIC_DIMENSIONS_ORDER_BY_END_TIME, entityClass)
-        .setParameter("collection", collection).setParameter("metric", metric)
-        .setParameter("dimensions", dimensions).setMaxResults(1).getSingleResult();
+    try {
+      return getEntityManager()
+          .createQuery(FIND_BY_COLLECTION_METRIC_DIMENSIONS_ORDER_BY_END_TIME, entityClass)
+          .setParameter("collection", collection).setParameter("metric", metric)
+          .setParameter("dimensions", dimensions).setMaxResults(1).getSingleResult();
+    } catch (NoResultException e) {
+      return null;
+    }
   }
 
   @Transactional
   public AnomalyMergedResult findLatestByFunctionIdDimensions(Long functionId, String dimensions) {
-    return getEntityManager().createQuery(FIND_BY_FUNCTION_DIMENSIONS, entityClass)
-        .setParameter("functionId", functionId).setParameter("dimensions", dimensions)
-        .setMaxResults(1).getSingleResult();
+    try {
+      return getEntityManager().createQuery(FIND_BY_FUNCTION_DIMENSIONS, entityClass)
+          .setParameter("functionId", functionId).setParameter("dimensions", dimensions)
+          .setMaxResults(1).getSingleResult();
+    } catch (NoResultException e) {
+      return null;
+    }
   }
 
   @Transactional
   public AnomalyMergedResult findLatestByFunctionIdOnly(Long functionId) {
```

Comment:
Would it help to have a findSingleResultByParams since this is being used in mulitple places